cdap-common/src/main/java/co/cask/cdap/common/logging/LocationAwareWrapperLogger.java

```diff
@@ -395,6 +412,13 @@ public void error(Marker marker, String msg, Throwable t) {
   @Override
   public void log(@Nullable Marker marker, String fqcn, int level,
                   String message, @Nullable Object[] args, @Nullable Throwable t) {
+    beforeLog();
```

Comment:
should this be in a try ... finally? Not sure whether log_internal can throw anything